### PR TITLE
Add RazorLangVersion 5.0

### PR DESCRIPTION
**Pull request by pranavkm**
I spoke to Taylor offline and it looks like we can get away without creating a new RazorConfiguration or extension for MVC. I did some testing with MVC 5.0 and 3.1 apps, and tooling appears to work correctly.

### CODE REVIEW

File: src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs [hunk 1]
```diff
@@ -625,13 +625,13 @@ public async Task Build_WithoutServer_ErrorDuringBuild_DisplaysErrorInMsBuildOut
         {
             var result = await DotnetMSBuild(
                 "Build",
-                "/p:UseRazorBuildServer=false /p:RazorLangVersion=5.0",
+                "/p:UseRazorBuildServer=false /p:RazorLangVersion=6.0",
```
Comment (SteveSandersonMS) [inline src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs:628]: Minor point, but why keep rolling this forward each time? What about:

```suggestion
                "/p:UseRazorBuildServer=false /p:RazorLangVersion=99999.0",
```

**Review by SteveSandersonMS — APPROVED** [review]
Thanks a lot for sorting this out, @pranavkm!

File: src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs (hunk 1 above)
Comment (pranavkm) [reply src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs:628]: I don't really know why this test is authored this way. @NTaylorMullen \ @ajaybhargavb ?

File: src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs [hunk 2]
```diff
@@ -625,28 +625,26 @@ public async Task Build_WithoutServer_ErrorDuringBuild_DisplaysErrorInMsBuildOut
...
                 "Build",
-                "/p:UseRazorBuildServer=false /p:RazorLangVersion=5.0",
+                "/p:UseRazorBuildServer=false /p:RazorLangVersion=99.0",
                 suppressBuildServer: true);
 
             Assert.BuildFailed(result);
             Assert.BuildOutputContainsLine(
                 result,
-                $"Invalid option 5.0 for Razor language version --version; must be Latest or a valid version in range 1.0 to 3.0.");
+                $"Invalid option 99.0 for Razor language version --version; must be Latest or a valid version in range 1.0 to 5.0.");
```
Comment (ajaybhargavb) [inline src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs:634]: > or a valid version in range 1.0 to 5.0.

This is a confusing statement indicating 4.0 might be valid. Could we rephrase this somehow?

**Review by ajaybhargavb — APPROVED** [review]

File: src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs (hunk 2 above)
Comment (pranavkm) [reply src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs:634]: Suggestions? I would note that none of the point values are supported. (e.g. 3.1 is not valid).

Comment (ajaybhargavb) [reply src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs:634]: If it is easy to do, we could just list all the possible supported values. Just like how dotnet lists all the available SDKs

File: src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs (hunk 1 above)
Comment (ajaybhargavb) [reply src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs]: No reason. I was probably being dumb.

File: src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs (hunk 2 above)
Comment (pranavkm) [reply src/Razor/Microsoft.NET.Sdk.Razor/integrationtests/BuildIntegrationTest.cs:634]: Here's what the compiler does: 

Error	CS1617	Invalid option '9.0' for /langversion. Use '/langversion:?' to list supported values

It's not really easy to find this. I'm kinda tempted to leave these be. AFAIK, nobody configures these. Plus the language versions are precariously tied to the runtime version that setting this is always going to get you in trouble.

File: src/Razor/Microsoft.AspNetCore.Razor.Language/src/RazorLanguageVersion.cs [hunk 3]
```diff
@@ -41,6 +43,11 @@ public static bool TryParse(string languageVersion, out RazorLanguageVersion ver
                 version = Experimental;
                 return true;
             }
+            else if (languageVersion == "5.0")
```
Comment (NTaylorMullen) [inline src/Razor/Microsoft.AspNetCore.Razor.Language/src/RazorLanguageVersion.cs:46]: You need to add this equivalent to the tooling repo too: https://github.com/dotnet/aspnetcore-tooling/blob/671d0792fb3fb587766a5800fafb0eee2d8f43cd/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/FallbackRazorConfiguration.cs#L37-L39

**Pull request merged**